core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalCheckpointThread.java

```diff
@@ -223,17 +230,11 @@ private void runInternal() {
             break;
           case LOG:
             entry = mJournalReader.getEntry();
-            try {
-              if (!mMaster.processJournalEntry(entry)) {
-                JournalUtils
-                    .handleJournalReplayFailure(LOG, null, "%s: Unrecognized journal entry: %s",
-                        mMaster.getName(), entry);
-              } else {
-                JournalUtils.sinkAppend(mJournalSinks, entry);
-              }
-            } catch (Throwable t) {
-              JournalUtils.handleJournalReplayFailure(LOG, t,
-                  "%s: Failed to read or process journal entry %s.", mMaster.getName(), entry);
+            if (mMaster.processJournalEntry(entry)) {
+              JournalUtils.sinkAppend(mJournalSinks, entry);
+            } else {
+              throw new RuntimeException(String.format("%s: Unrecognized journal entry: %s",
```

Comment:
We don't plan to tolerate journal replay failure here?
So when we init from backup -> allow tolerating journal replay failure
when playing checkpoint -> do not allow tolerating journal replay failure

would it be better to not tolerate at all or allow failure in both cases